bcgov/biohubbc
# BHBC-2215: Update Project Role permissions for API

# Overview

## Links to Jira tickets

[BHBC-2215](https://quartech.atlassian.net/browse/BHBC-2215?atlOrigin=eyJpIjoiYjYxZmMzZjA1YzcwNDFhZjliNmU1NDg3M2RlZTg2MjUiLCJwIjoiaiJ9)

## Description of relevant changes

- updated view occurrences endpoint to properly check authorization rules
- updated various endpoints to keep inline with current role rules (found in confluence SIMS>Roles & Permissions)

## Testing Procedure
Viewer Case
- Create a project/ survey as an admin
- Invite a participant with the viewer role
- Login as that newly invited user 
- Interact with the project/ survey 
- attempt to save or modify metadata, attachments and be met with errors / access denied messages

Editor Case
- Create a project/ survey as an admin
- Invite a participant with the editor role
- Login as the newly invited user
- Interact with the project/ survey
- The editor should be able to edit metadata, upload/ delete attachments and even delete the survey itself

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code


### Style


### Documentation


### Tests


### Linting/Formatting

       _See the `lint` commands in package.json_
       _See the `format` commands in package.json_

### SonarCloud
